classes/war/WarAction.php

```diff
@@ -80,24 +73,42 @@ class Operation {
     public int $last_update_ms;
     public int $interval_progress = 0;
 
-    public function __construct(System $system, User $user, array $operation_data)
+    public function __construct(System $system, User $user, array $war_action_data)
     {
-        foreach ($operation_data as $key => $value) {
+        foreach ($war_action_data as $key => $value) {
             $this->$key = $value;
         }
         $this->system = $system;
         $this->user = $user;
-        if ($this->type == self::OPERATION_LOOT || $this->type == self::OPERATION_LOOT_TOWN) {
-            $interval = self::LOOT_OPERATION_INTERVAL_SECONDS;
-        } else {
-            $interval = self::BASE_OPERATION_INTERVAL_SECONDS;
+        switch ($this->type) {
+            case self::WAR_ACTION_LOOT:
+            case self::WAR_ACTION_LOOT_TOWN:
+                $interval = self::LOOT_WAR_ACTION_INTERVAL_SECONDS;
+                break;
+            case self::WAR_ACTION_INFILTRATE:
+                $interval = self::BASE_WAR_ACTION_INTERVAL_SECONDS;
+                $interval = $interval * (100 / (100 + $this->user->village->policy->infiltrate_speed));
+                $interval = round($interval * (100 / (100 + $this->user->village->active_upgrade_effects[VillageUpgradeConfig::UPGRADE_EFFECT_INFILTRATE_SPEED])), 1);
```

Comment:
This logic is duplicated here and `progressActiveWarAction`, could consider pulling it to its own method so we have a single source of truth for "what is the interval/cost/speed for this type of war action"